### PR TITLE
Hashed password chat

### PR DESCRIPTION
**Pull request by livlaming**
sorry I forgot the Li/ before the branch name!
small pr :)!

### CODE REVIEW

File: backend/src/chats/chat/chat.controller.ts [hunk 1]
```diff
@@ -12,14 +12,18 @@ import { ChatService } from './chat.service';
...
 	constructor(private readonly chatService: ChatService) {}
 
 	@Post()
-	create(@Body() createChatDto: CreateChatDto) {
-		// use bcrypt to hash passwords!
+	async create(@Body() createChatDto: CreateChatDto) {
+		if (createChatDto.hasOwnProperty('password')) {
+			const hashed = await bcrypt.hash(createChatDto.password, 11);
+			createChatDto.password = hashed;
+		}
```
Comment (Ryno95) [inline backend/src/chats/chat/chat.controller.ts:26]: Same as  #102 not possible to create a chat without a password, in DTO = optional in entity mandatory. 
Do we want to be able to make chats without a pw?  If so we can add it as an optional in the entity.

**Review by Ryno95 — COMMENTED** [review]
Looks good, only comment is the same as in #102 :)

**Review by Ryno95 — COMMENTED** [review]
Also, there is no validation for the visibility at this point, if we are doing private/public, you can use an enum  or something like this: 
``` typescript
const validVisibility = ['public', 'private'];

export class Chat {

    @IsIn(validVisibility)
  visibility: string
}
```

File: backend/src/chats/chat/chat.controller.ts (hunk 1 above)
Comment (vincentvis) [reply backend/src/chats/chat/chat.controller.ts:26]: this is a better spot for this xD But I had already responded in the other PR as well.
but copy pasta here for completeness sake

Changing the entity password to @Column({ nullable: true }) solves this

File: backend/test/chat.e2e-spec.ts [hunk 2]
```diff
@@ -5,10 +5,21 @@ import { ChatController } from '../src/chats/chat/chat.controller';
...
 describe('chat e2e', () => {
 	let app: INestApplication;
 	let chatController: ChatController;
+	let messageService: MessageService;
+	let messageController: MessageController;
+
+	const testMessages: CreateMessageDto[] = [
+		{ sender_id: 1, chat_id: 1, content: 'hello' },
+		{ sender_id: 2, chat_id: 1, content: 'world!' },
+	];
```
Comment (vincentvis) [inline backend/test/chat.e2e-spec.ts:21]: This is added but not used in the e2e test

(same for MockChat a few lines below this)

**Review by vincentvis — CHANGES_REQUESTED** [review]
Needs an update to allow chats without password, but other than that good :)

File: backend/src/chats/chat/chat.controller.ts (hunk 1 above)
Comment (livlaming) [reply backend/src/chats/chat/chat.controller.ts:26]: alright!

**Review by vincentvis — APPROVED** [review]

**Review by Ryno95 — APPROVED** [review]
Nois! MEEEERGE THAT BRANCH!!! :D

**Pull request merged**